Source/DafnyCore/Verifier/BoogieGenerator.Methods.cs

```diff
@@ -1876,6 +1876,10 @@ private Boogie.Procedure AddMethod(Method m, MethodTranslationKind kind) {
       }
     }
 
+    public static bool ModifiesClauseIsEmpty(Specification<FrameExpression> modifiesClause) {
+      return modifiesClause.Expressions.All(e => e.E is SetDisplayExpr setDisplayExpr && !setDisplayExpr.Elements.Any());
+    }
```

Comment:
`opaque modifies { {} }`

that's not valid Dafny syntax.

What's wrong with just supporting `modifies {}` ?

Reply:
I tied it in the IDE and it seemed to work.
I tend to find semantic rather than syntactic rules more natural, because I find them easier to explain.

Reply:
> I tend to find semantic rather than syntactic rules more natural, because I find them easier to explain.

That's fair. I think there could be a catch22 here, since to evaluate if an arbitrary set expression evaluates to empty would require the solver or something similar, but the only point of doing this evaluation is to improve solver performance.